app/graphql/scripts.js

```diff
@@ -0,0 +1,83 @@
+const fs = require('fs'),
+  path = require('path');
+
+const INDEX_FILE = 'index.js',
+  SCRIPTS_FILE = 'scripts.js';
+
+const generateIndex = dirent =>
+  fs.readdirSync(path.join(__dirname, dirent.name)).reduce(
+    (singleModule, file) => {
+      const fileName = file.replace(/.js/gi, '');
+      const definitions = require(path.join(__dirname, dirent.name, fileName));
+      singleModule[fileName] = definitions[fileName] ? definitions[fileName] : definitions;
+      if (definitions.schema) {
+        singleModule.schemas.push(definitions.schema);
+      }
+      return singleModule;
+    },
+    { schemas: [] }
+  );
+
+exports.importEverything = () =>
+  fs.readdirSync(__dirname, { withFileTypes: true }).reduce(
+    (imports, dirent) => {
+      if (dirent.isDirectory()) {
+        const index = generateIndex(dirent);
+        imports.modules.push(index);
+        return imports;
+      } else if (dirent.name === INDEX_FILE || dirent.name === SCRIPTS_FILE) {
+        return imports;
+      }
+      imports[dirent.name.replace(/.js/gi, '')] = require(path.join(__dirname, dirent.name));
+      return imports;
+    },
+    { modules: [] }
+  );
+
+exports.destructureModules = modules => {
+  const rootObject = { Query: {}, Mutation: {}, Subscription: {} };
+  return modules.reduce(
+    (destructuredModules, currentModule) => {
+      destructuredModules.schemas.push(...currentModule.schemas);
+
+      const { fieldResolvers } = currentModule.resolvers ? currentModule.resolvers : {};
+      destructuredModules.resolvers = {
+        ...destructuredModules.resolvers,
+        Query: {
+          ...destructuredModules.resolvers.Query,
+          ...currentModule.queries
+        },
+        Mutation: {
+          ...destructuredModules.resolvers.Mutation,
+          ...currentModule.mutations
+        },
+        Subscription: {
+          ...destructuredModules.resolvers.Subscription,
+          ...currentModule.subscriptions
+        },
+        ...fieldResolvers
```

Comment:
Should we add type-resolvers here?

Reply:
I named `fieldResolvers`? Do you think I should change the name?

Reply:
I think they are different things. I see it this way:
If I have this types: `a: [type1], b: [type1]`, a fieldResolver would refer to `a` or `b`, while typeResolver would refer to `type1`.
What do you think?